From 235ed1dc43f85622684696898779d8e2c1f2815c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:58 -0400
Subject: [usb] kobil_sct: fix control requests without data stage

Message-id: <1410887463-20674-467-git-send-email-dzickus@redhat.com>
Patchwork-id: 95618
O-Subject: [RHEL7 PATCH 466/591] USB: kobil_sct: fix control requests without data stage
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 90419cfcb5d9c889b10dc51363c56a4d394d670e
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:35 2014 +0200

    USB: kobil_sct: fix control requests without data stage

    Fix incorrect pipe directions and remove bogus data buffer arguments
    from control requests without data stage.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/kobil_sct.c b/drivers/usb/serial/kobil_sct.c
index 0f5f09c..44ab6bb 100644
--- a/drivers/usb/serial/kobil_sct.c
+++ b/drivers/usb/serial/kobil_sct.c
@@ -215,13 +215,13 @@ static int kobil_open(struct tty_struct *tty, struct usb_serial_port *port)
    priv->device_type == KOBIL_ADAPTER_K_PRODUCT_ID) {
   /* Setting Baudrate, Parity and Stopbits */
   result = usb_control_msg(port->serial->dev,
-     usb_rcvctrlpipe(port->serial->dev, 0),
+     usb_sndctrlpipe(port->serial->dev, 0),
      SUSBCRequest_SetBaudRateParityAndStopBits,
      USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
      SUSBCR_SBR_9600 | SUSBCR_SPASB_EvenParity |
        SUSBCR_SPASB_1StopBit,
      0,
-     transfer_buffer,
+     NULL,
      0,
      KOBIL_TIMEOUT
   );
@@ -229,12 +229,12 @@ static int kobil_open(struct tty_struct *tty, struct usb_serial_port *port)
 
   /* reset all queues */
   result = usb_control_msg(port->serial->dev,
-     usb_rcvctrlpipe(port->serial->dev, 0),
+     usb_sndctrlpipe(port->serial->dev, 0),
      SUSBCRequest_Misc,
      USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
      SUSBCR_MSC_ResetAllQueues,
      0,
-     transfer_buffer,
+     NULL,
      0,
      KOBIL_TIMEOUT
   );
@@ -445,12 +445,12 @@ static int kobil_tiocmset(struct tty_struct *tty,
   else
    dev_dbg(dev, "%s - Clearing DTR\n", __func__);
   result = usb_control_msg(port->serial->dev,
-     usb_rcvctrlpipe(port->serial->dev, 0),
+     usb_sndctrlpipe(port->serial->dev, 0),
      SUSBCRequest_SetStatusLinesOrQueues,
      USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
      ((dtr != 0) ? SUSBCR_SSL_SETDTR : SUSBCR_SSL_CLRDTR),
      0,
-     transfer_buffer,
+     NULL,
      0,
      KOBIL_TIMEOUT);
  } else {
@@ -459,12 +459,12 @@ static int kobil_tiocmset(struct tty_struct *tty,
   else
    dev_dbg(dev, "%s - Clearing RTS\n", __func__);
   result = usb_control_msg(port->serial->dev,
-   usb_rcvctrlpipe(port->serial->dev, 0),
+   usb_sndctrlpipe(port->serial->dev, 0),
    SUSBCRequest_SetStatusLinesOrQueues,
    USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
    ((rts != 0) ? SUSBCR_SSL_SETRTS : SUSBCR_SSL_CLRRTS),
    0,
-   transfer_buffer,
+   NULL,
    0,
    KOBIL_TIMEOUT);
  }
@@ -514,7 +514,7 @@ static void kobil_set_termios(struct tty_struct *tty,
  tty_encode_baud_rate(tty, speed, speed);
 
  result = usb_control_msg(port->serial->dev,
-    usb_rcvctrlpipe(port->serial->dev, 0),
+    usb_sndctrlpipe(port->serial->dev, 0),
     SUSBCRequest_SetBaudRateParityAndStopBits,
     USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
     urb_val,
@@ -546,12 +546,12 @@ static int kobil_ioctl(struct tty_struct *tty,
    return -ENOBUFS;
 
   result = usb_control_msg(port->serial->dev,
-     usb_rcvctrlpipe(port->serial->dev, 0),
+     usb_sndctrlpipe(port->serial->dev, 0),
      SUSBCRequest_Misc,
      USB_TYPE_VENDOR | USB_RECIP_ENDPOINT | USB_DIR_OUT,
      SUSBCR_MSC_ResetAllQueues,
      0,
-     NULL, /* transfer_buffer, */
+     NULL,
      0,
      KOBIL_TIMEOUT
    );
-- 
1.7.1